File: backend/app/schemas/confirmations.py
from datetime import datetime

from pydantic import BaseModel, Field


class ConfirmationRequestRead(BaseModel):
    id: str
    confirmation_type: str
    status: str
    doctor_id: str
    notification_id: str | None
    assignment_id: str | None
    mission_id: str | None
    due_at: datetime | None
    responded_at: datetime | None
    response_channel: str | None
    created_by: str | None
    created_at: datetime
    updated_at: datetime

    model_config = {"from_attributes": True}


class ConfirmationRequestListResponse(BaseModel):
    items: list[ConfirmationRequestRead]
    total: int


class ProcessOverdueConfirmationsResponse(BaseModel):
    expired: int
    alerts_created: int


class PublicConfirmationRead(BaseModel):
    confirmation_type: str
    status: str
    doctor_name: str | None
    due_at: datetime | None
    responded_at: datetime | None


class PublicConfirmationResponse(BaseModel):
    confirmation_type: str
    status: str
    doctor_name: str | None
    responded_at: datetime | None


class PublicConfirmationActionRequest(BaseModel):
    note: str | None = Field(default=None, max_length=500)

File: frontend/src/api/confirmations.ts
import { apiFetch } from "./client";

export interface ProcessOverdueConfirmationsResponse {
  expired: number;
  alerts_created: number;
}

export interface PublicConfirmationRead {
  confirmation_type: string;
  status: string;
  doctor_name: string | null;
  due_at: string | null;
  responded_at: string | null;
}

export interface PublicConfirmationResponse {
  confirmation_type: string;
  status: string;
  doctor_name: string | null;
  responded_at: string | null;
}

export interface ConfirmationRequestRead {
  id: string;
  confirmation_type: string;
  status: string;
  doctor_id: string;
  doctor_name?: string;
  assignment_id: string | null;
  due_at: string | null;
  responded_at: string | null;
  response_channel: string | null;
  response_payload?: Record<string, unknown> | null;
  created_at: string;
  escalated_at: string | null;
}

export interface ConfirmationRequestListResponse {
  items: ConfirmationRequestRead[];
  total: number;
}

export const confirmationsApi = {
  list: (params?: { status?: string; doctor_id?: string; limit?: number }) => {
    const q = new URLSearchParams();
    if (params?.status) q.set("status", params.status);
    if (params?.doctor_id) q.set("doctor_id", params.doctor_id);
    if (params?.limit) q.set("limit", String(params.limit));
    return apiFetch<ConfirmationRequestListResponse>(`/confirmation-requests?${q}`);
  },

  processOverdue: () =>
    apiFetch<ProcessOverdueConfirmationsResponse>(
      "/confirmation-requests/process-overdue",
      { method: "POST" },
    ),

  getPublicConfirmation: (token: string) =>
    apiFetch<PublicConfirmationRead>(
      `/confirmation-requests/public/${encodeURIComponent(token)}`,
    ),

  respondPublicConfirmation: (token: string, action: "received" | "confirm", note?: string) =>
    apiFetch<PublicConfirmationResponse>(
      `/confirmation-requests/public/${encodeURIComponent(token)}/${action}`,
      {
        method: "POST",
        body: JSON.stringify({ note: note?.trim() || null }),
      },
    ),
};
